assets/src/components/inputText/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Muriel-styled Text/Number Input.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { TextControl, withFocusOutside } from '@wordpress/components';
+import { Component } from '@wordpress/element';
+
+const InputText = withFocusOutside(
+	class extends Component {
+		/**
+		 * Constructor.
+		 */
+		constructor( props ) {
+			super( props  );
+			this.state = {
+				value: props.value,
+				isFocused: false
+			};
+		};
+
+		/**
+		 * withFocusOutside: handles component onClick.
+		 */
+		handleFocusOutside() {
+			this.setState( { isFocused: false } );
+		}
+
+		/**
+		 * Handle component onClick.
+		 */
+		handleOnClick = () => {
+			this.setState( { isFocused: true } );
+		};
+
+		/**
+		 * Handle component onChange.
+		 */
+		handleOnChange = value => {
+			this.setState( { value } );
+		};
+
+		/**
+		 * Get component's className depending on state and context.
+		 */
+		getTextInputClassName = ( disabled, isEmpty, isActive ) => {
+			let className = "with-value";
+			if ( disabled ) {
+				className = "disabled";
+			} else if ( isEmpty ) {
+				className = "empty";
+			} else if ( isActive ) {
+				className = "active";
+			}
+
+			return className;
+		};
+
+		/**
+		 * Render.
+		 */
+		render() {
+			const { value, isFocused } = this.state;
+			const { label, disabled } = this.props;
+			const isEmpty = ! value;
+			const isActive = isFocused && ! disabled;
+			const classNameInput= this.getTextInputClassName( disabled, isEmpty, isActive );
+
+			return (
+				<TextControl
+					className={ "newspack-input-text " + classNameInput }
+					placeholder={ label }
```

Comment:
Wouldn't there be a good argument for distinct `label` and `placeholder` values? If so, we might want to remove this and leave it to the parent to define, if desired?

Reply:
If I'm not mistaken, the Muriel implementation uses only one of them (label or placeholder), not both:
![Muriel Input Control](https://user-images.githubusercontent.com/7317227/56913019-75140c00-6a65-11e9-812e-dc16a75edbed.png "Muriel Input Control")

The way I saw it:
- in the 2nd example on the image, we use the <label> above the <input> (which are children of the TextComponent from @wordpress/components)
- and to implement the 1st and 3rd examples, I've hidden the <label> and set the placeholder on the <input>. But I might have taken a wrong approach there, and maybe should have enlarged the <label> and hidden the <input> (in which case the placeholder prop would not be used). If you believe this should be changed, please just comment and I will. At this point, it seems to me that the effect is exactly the same, but it might be semantically cleaner.

Reply:
I see what you're saying. Let's put a pin in this and revisit in a later PR when we have determined if the styling here is in line with the Muriel approach Sonja had in mind.